external/importL10n/locales.mjs

```diff
@@ -28,41 +27,32 @@ function normalizeText(s) {
   return s.replaceAll(/\r\n?/g, "\n").replaceAll("\uFEFF", "");
 }
 
-function downloadLanguageCodes() {
+async function downloadLanguageCodes() {
   console.log("Downloading language codes...\n");
 
   const ALL_LOCALES =
     "https://hg.mozilla.org/mozilla-central/raw-file/tip/browser/locales/all-locales";
 
-  return new Promise(function (resolve) {
-    https.get(ALL_LOCALES, function (response) {
-      if (response.statusCode === 200) {
-        let content = "";
-        response.setEncoding("utf8");
-        response.on("data", function (chunk) {
-          content += chunk;
-        });
-        response.on("end", function () {
-          content = content.trim(); // Remove any leading/trailing white-space.
-          const langCodes = normalizeText(content).split("\n");
-          // Remove all locales that we don't want to download below.
-          for (const langCode of [DEFAULT_LOCALE, ...EXCLUDE_LANG_CODES]) {
-            const i = langCodes.indexOf(langCode);
-            if (i > -1) {
-              langCodes.splice(i, 1);
-            }
-          }
-          resolve(langCodes);
-        });
-      } else {
-        resolve([]);
-      }
-    });
-  });
+  const response = await fetch(ALL_LOCALES);
+  if (!response.ok) {
+    throw new Error(response.statusText);
+  }
+  const content = await response.text();
+
+  // Remove any leading/trailing white-space.
+  const langCodes = normalizeText(content.trim()).split("\n");
+  // Remove all locales that we don't want to download below.
+  for (const langCode of [DEFAULT_LOCALE, ...EXCLUDE_LANG_CODES]) {
+    const i = langCodes.indexOf(langCode);
+    if (i > -1) {
+      langCodes.splice(i, 1);
+    }
+  }
+  return langCodes;
```

Comment:
```suggestion
  // Remove all locales that we don't want to download below.
  return langCodes.filter(langCode =>
  	langCode !== DEAULT_LOCALE && !EXCLUDE_LANG_CODES.includes(langCode)
  );
```

or even better, make `EXCLUDE_LANG_CODES` a `Set` and use `.has` instead of `includes`.

Using `.splice` needs to re-iterate `langCodes` every time to move back all its elements by one, while with this `langCodes` is only iterated once.

Reply:
Sure, we can do instead; thank you!

---

*Just in case it's not obvious:* Note that this code isn't run automatically in any part of the build process.
Instead a contributor will run this *manually* every once in a while, in order to update the l10n files.